### PR TITLE
Magento to Dynamics365 - Orders

### LINKED ISSUE #95: Magento to Dynamics365 - Orders
Sync order details (order ID, customer info, items, total price, order status) from Magento to Dynamics 365.

Use our dynamics connector.

```http
GET /rest/V1/orders
Host: your-magento-store.com
Authorization: Bearer your-magento-access-token
```

```http
POST /api/data/v9.1/salesorders
Host: your-dynamics-instance.api.crm.dynamics.com
Authorization: Bearer your-dynamics-access-token
Content-Type: application/json

{
  "name": "Order 1001",
}
```

### PR DESCRIPTION
**Pull request by jannejjj**
closes #95 

Checklist for review:
- [x] Process name is fine, uses normal casing and descriptive
  - [x] If applicable, the process name follows the naming convention "{source/format} to {destination/format} - {object}"
- [x] Process description is fine, uses normal casing, dots and is descriptive
  - [x] Element names use normal casing, no unneeded capital letters
- [x] Correct casing used for formats and terms, e.g. CSV, FTP, JSON, Avro
  - [x] Element names do not contain dots
  - [x] Comments / note element text are in good language and end with dots
- [x] Template parameters checked
  - [x] Parameter follows the CamelCase naming convention, abbreviations included, e.g. ConnectionString, BaseUrl, ApiKey
  - [x] Parameter descriptions are good and end with dots
  - [x] Sensitive parameters are marked as secrets
- [x] "Turn off logging" switch is on for any task that has sensitive data as output, e.g. tokens in the header.
- [x] Process end points like error throws and return statements have titles (double-click) and reasonable information.
  - [x] Results have good summaries, preferably with statistics of some sort like "processed X files"
  - [x] Errors have some error info, usually from task (e.g. some ErrorMessage property contains additional error info)
  - [x] Errors are pointed downwards from their respective source, e.g. an exclusive decision
- [x] Process is organised with groups (dashed boxes)
- [x] Straight lines are preferred where possible
- [x] Subprocesses are forbidden
- [x] Text mode is preferred over Expressions in expressions, e.g. instead of *$"This is {#var.error}"* use *This is {{#var.error}}*
- [x] Process metadata file is present and content is fine (version number and name)
- [x] Process export file is present (process.json)
- [x] Process description file is present (all sections filled in)
- [x] Process description has a SVG screenshot of the process
- [x] Template files are going to the correct folder

## Summary by CodeRabbit

- **New Features**
	- Introduced a template for synchronizing orders from Magento to Dynamics365.
	- Added detailed documentation outlining prerequisites, implementation, usage, error handling, and considerations for unit of measurement and price list in Dynamics365.

### CODE REVIEW

File: Templates/E-Commerce/Magento/Magento to Dynamics365 - Orders/long-description.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# Implementation and Usage Notes
+
+To insert orders into Dynamics365, a unit of measurement has to be defined for the products in the orders. By default, this process uses a unit of measurement called "Primary Unit", but this behavior can be changed to another unit in the corresponding task. In any case, the desired unit of measurement, be it the default Primary Unit or another one, has to exist in Dynamics365.
+
+The products in the orders to be inserted must also be in a Price List on Dynamics365, to correctly determine their price. This template assumes that the products are already in a price list. If you have imported the products from Magento to Dynamics365 using the template, the products should already be in a price list called "Magento Price List".
+
+The orders inserted into Dynamics365 using this template have an ID starting with "MAG" (as in Magento) followed by their ID from Magento, e.g., "MAG11". Similarly, the names of the imported orders begin with "Magento order" followed by their ID from Magento, e.g., "Magento Order 10".
+
+Existing orders are not updated by this template. Whether an order already exists is checked by comparing the IDs of the orders.
```
Comment (FrendsSarlinS) [inline Templates/E-Commerce/Magento/Magento to Dynamics365 - Orders/long-description.md:23]: Can you be a bit more descriptive in the usage notes, as in how the process works? Mostly it's now opening the prerequisites a bit further. What information is retrieved, in which order, and which fields are matched to check if an object is in Dynamics365?

**Review by FrendsSarlinS — COMMENTED** [review]
1) Process description seems to have vanished. It happens to me sometimes as well. 😄 

2) Are the throws inside the second foreach loop intentional, or should they be returns? For example, if an order is created but the product doesn't exist, all orders after that order are not handled, and the process needs to be run again.

3) One error return has the title "Move to next order" as opposed to others, yet it also adds a message to the error variable.

File: Templates/E-Commerce/Magento/Magento to Dynamics365 - Orders/long-description.md (hunk 1 above)
Comment (jannejjj) [reply Templates/E-Commerce/Magento/Magento to Dynamics365 - Orders/long-description.md:23]: Yep, sure! I'll change it a bit 😄

**Comment by jannejjj** [comment]
Re-added the process description, and changed how the logic for inserting the lines into the orders / error handling works.

File: Templates/E-Commerce/Magento/Magento to Dynamics365 - Orders/process.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+			"Version": 1,
+			"UniqueIdentifier": "986a6287-7c3f-4b21-8aec-e591682c4c98",
+			"ProcessVariablesJson": "{\"Dynamics365ClientId\":{\"Value\":\"\\\"aaaaa-bbbbb-ccccc-ddddd\\\"\",\"IsSecret\":false,\"Mode\":\"text\",\"Description\":\"The client ID for Dynamics365.\"},\"Dynamics365TenantId\":{\"Value\":\"\\\"11111-22222-33333-44444\\\"\",\"IsSecret\":false,\"Mode\":\"text\",\"Description\":\"The tenant ID for Dynamics365.\"},\"Dynamics365ClientSecret\":{\"Value\":\"\\\"\\\"\",\"IsSecret\":true,\"Mode\":\"text\",\"Description\":\"The client secret for Dynamics365.\"},\"Dynamics365Url\":{\"Value\":\"\\\"https://<my-dynamics-instance>.dynamics.com\\\"\",\"IsSecret\":false,\"Mode\":\"text\",\"Description\":\"The URL to for Dynamics365 instance.\"},\"MagentoBaseUrl\":{\"Value\":\"\\\"\\\"\",\"IsSecret\":false,\"Mode\":\"text\",\"Description\":\"The base URL for Magento.\"},\"MagentoUser\":{\"Value\":\"\\\"myUsername\\\"\",\"IsSecret\":false,\"Mode\":\"text\",\"Description\":\"The username for Magento.\"},\"MagentoPassword\":{\"Value\":\"\\\"myPassword\\\"\",\"IsSecret\":true,\"Mode\":\"text\",\"Description\":\"The password for your Magento user.\"}}",
```
[the remark this answers is not in the document]
Comment (jannejjj) [reply Templates/E-Commerce/Magento/Magento to Dynamics365 - Orders/process.json:15]: Those are placeholders, not the actual values.

**Review by FrendsSarlinS — APPROVED** [review]

**Pull request merged**